alphadia/fdrx/stats.py

```diff
@@ -0,0 +1,166 @@
+# make it an abc
+import numpy as np
+import pandas as pd
+from typing import List
+from alphadia.fragcomp import FragmentCompetition
+
+
+def _get_pep(
+    psm_df: pd.DataFrame,
+    score_column: str = "decoy_proba",
+    decoy_column: str = "decoy",
+    score_std: float = 0.01,
+    pep_granularity: int = 1000,
+    kernel_size: int = 20,
+):
+    """Implementation of a very simple nonparametric PEP estimation using a gaussian kernel.
+
+    Parameters
+    ----------
+
+    psm_df : pd.DataFrame
+        The dataframe containing the PSMs.
+
+    score_column : str, default='decoy_proba'
+        The name of the column containing the score to use for the selection.
+
+    decoy_column : str, default='decoy'
+        The name of the column containing the decoy information.
+
+    score_std : float, default=0.01
+        The standard deviation of the gaussian kernel.
+
+    pep_granularity : int, default=1000
+        The number of bins to use for the score histogram.
+
+    kernel_size : int, default=20
+        The size of the kernel to use for the convolution.
+
+    Returns
+    -------
+
+    np.ndarray
+        The PEP values with same shape and order as the input dataframe.
+
+    """
+
+    score_bins = np.linspace(0, 1, pep_granularity)
+    target_decoy = psm_df[decoy_column].values
+    score = psm_df[score_column].values
+
+    target_hist, _ = np.histogram(score[target_decoy == 0], bins=score_bins)
+    decoy_hist, _ = np.histogram(score[target_decoy == 1], bins=score_bins)
+
+    std_norm = score_std / (score_bins[1] - score_bins[0])
+    # create a gaussian kernel of 0.01 width with 5 elements
+    kernel_gaussian = np.exp(
+        -(np.arange(-kernel_size, kernel_size + 1) ** 2) / (2 * std_norm**2)
+    )
+
+    # convolve the target and decoy histograms with the kernel
+    target_hist = np.convolve(target_hist, kernel_gaussian, mode="same")
+    decoy_hist = np.convolve(decoy_hist, kernel_gaussian, mode="same")
+
+    # numerical stability
+    EPSILON = 1e-6
+    pep = decoy_hist / (target_hist + decoy_hist + EPSILON)
+
+    return pep[np.digitize(score, score_bins) - 1]
+
+
+def _get_q_values(
```

Comment:
would "add_q_values" be more descriptive?

Reply:
Good point, add sounds more inplace